src/main/java/com/mucommander/ui/action/impl/ToggleSinglePanelAction.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ * Copyright (C) 2002-2012 Maxence Bernard
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.ui.action.impl;
+
+import java.util.Map;
+
+import javax.swing.KeyStroke;
+
+import com.mucommander.ShutdownHook;
+import com.mucommander.ui.action.AbstractActionDescriptor;
+import com.mucommander.ui.action.ActionCategories;
+import com.mucommander.ui.action.ActionCategory;
+import com.mucommander.ui.action.ActionDescriptor;
+import com.mucommander.ui.action.ActionFactory;
+import com.mucommander.ui.action.MuAction;
+import com.mucommander.ui.main.MainFrame;
+
+/**
+ * Toggles isVisible state of the right panel, imitating single/two panel view switch.
+ */
+public class ToggleSinglePanelAction extends MuAction {
+
+    private static final float TWO_PANELS_DEFAULT_RATIO = 0.5f;
+    private float previousRatio = TWO_PANELS_DEFAULT_RATIO;
+
+    ToggleSinglePanelAction(MainFrame mainFrame, Map<String, Object> properties) {
+        super(mainFrame, properties);
+    }
+
+    private void hideOtherPanel() {
+        mainFrame.getRightPanel().setVisible(false);
+    }
+
+    private void showOtherPanel() {
+        mainFrame.getRightPanel().setVisible(true);
```

Comment:
why is it always the right panel? don't we want to determine it according to the selected panel?